jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java

```diff
@@ -77,13 +78,18 @@ public JobHandle scheduleJob(Job job, JobContext ctx, Trigger trigger) {
 		// check if given timer job is marked as new timer meaning it was never scheduled before, 
 		// if so skip the check by timer name as it has no way to exist
 		if (!ctx.isNew()) {
-		    jobInstance = getTimerJobInstance(jobName);
-		    if (jobInstance == null) {
-		        jobInstance = scheduler.getTimerByName(jobName);
+            try {
+                jobInstance = getTimerJobInstance(jobName);
+                if (jobInstance == null) {
+                    jobInstance = scheduler.getTimerByName(jobName);
+                }
+            } catch (InvalidEJBTimer ex) {
+                logger.info(
+                        "Avoiding searching all timers since the timer mapping info pointed to a timer that is not longer valid");
```

Comment:
I'd prefer to have this logged at debug level and also have the jobName added so that the message provides more context